packages/stark-ui/src/modules/slider/components/slider.component.spec.ts

```diff
@@ -0,0 +1,186 @@
+		 *
+		 * The solution
+		 *   Putting a "stub" method on the "attachSliderInstanceUpdateHandler()" method, blocks the original code from being executed.
+		 *   Before testing the call of the "component.slider.on" method, put a "callThrough" on the spy and execute the "attachSliderInstanceUpdateHandler()" method manually.
+		 */
+		attachSliderInstanceUpdateHandlerSpy = spyOn(component, "attachSliderInstanceUpdateHandler").and.stub();
+
+		/***
+		 * fixture.detectChanges() triggers the ngOnInit() and ngAfterViewInit() methods in the component.
+		 */
+		hostFixture.detectChanges();
+
+		/***
+		 * The slider is created after the fixture.detectChanges() has been called
+		 */
+		spyOn(component.slider, "on").and.callThrough();
+		spyOn(component.slider, "set").and.callThrough();
+
+		mockChanged.subscribe();
+	});
+
+	describe("on initialization", () => {
+		it("should set internal component properties", () => {
+			expect(hostFixture).toBeDefined();
+			expect(component).toBeDefined();
+
+			expect(component.logger).not.toBeNull();
+			expect(component.logger).toBeDefined();
+			expect(component.routingService).not.toBeNull();
+			expect(component.routingService).toBeDefined();
+		});
+
+		it("should have inputs", () => {
+			expect(component.values).toBe(mockValues);
+			expect(component.sliderConfig).toBe(mockConfig);
+			expect(component.sliderId).toBe(mockSliderId);
+			expect(component.changed).toBeDefined();
+		});
+	});
+
+	describe("createSliderInstance", () => {
+		it("should create the slider instance and assign it to the internal variable", () => {
+			expect(component.noUiSliderLibrary.create).toHaveBeenCalledTimes(1);
+			expect(component.slider).toBeDefined();
+		});
+	});
+
+	describe("attachSliderInstanceUpdateHandler", () => {
+		it("should add an 'update' event listener to the slider instance", () => {
+			attachSliderInstanceUpdateHandlerSpy.and.callThrough();
+
+			component.attachSliderInstanceUpdateHandler();
+
+			expect(component.slider.on).toHaveBeenCalledTimes(1);
+
+			expect((<Spy>component.slider.on).calls.argsFor(0)[0]).toBe("update");
+			expect(typeof (<Spy>component.slider.on).calls.argsFor(0)[1]).toBe("function");
+		});
+
+		it("should update the values and call changed() whenever the update handler is triggered and the slider unencodedValues changed", () => {
+			hostComponent.sliderValues = newMockValues;
+			hostFixture.detectChanges();
+
+			expect(component.updateSliderInstanceValues).toHaveBeenCalledTimes(1);
+			expect(component.values).toBe(newMockValues);
+		});
+
+		it("should NOT do anything in case the update handler is triggered but the slider unencodedValues did not change", () => {
+			hostComponent.sliderValues = mockValues;
+			hostFixture.detectChanges();
+
+			expect(component.updateSliderInstanceValues).toHaveBeenCalledTimes(0);
+			expect(component.values).toBe(mockValues);
```

Comment:
This test is not complete... look at the test in the old Stark to see the rest of this test